datalake-api/src/main/java/com/sequenceiq/sdx/api/model/SdxClusterRequest.java

```diff
@@ -29,6 +30,8 @@
 
     private boolean enableRangerRaz;
 
+    private ImageSettingsV4Request imageSettingsV4Request;
```

Comment:
Please create a separate endpoint with a request like "CustomSdxClusterRequest". The runtime version is meaningless for this request, so we can create an ancestor class for SdxClusterRequest and CustomSdxClusterRequest without runtime version field. There is some logic for runtime version and it should be reviewed.

Reply:
I am also surprised that sdx-api is able reference classes from core-api

Reply:
introducing a base class, which does not have "runtime" field requires changing a lot of places. So I derive the SdxCustomClusterRequest from SdxClusterRequest and validate that the runtime field and image field cannot be set at the same time.